packages/amplify-category-api/src/category-utils/context-util.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  $TSAny,
+  $TSContext,
+  AmplifyCategories,
+  pathManager,
+} from 'amplify-cli-core';
+import path from 'path';
+import { PROVIDER_NAME } from '../graphql-transformer/constants';
+
+/**
+ * ContextUtil
+ * Some values are calculated on the basis of the context and options that come
+ * from the Amplify CLI, this class/singleton help calculate and cache those values
+ * for reference
+ */
+export class ContextUtil {
+  private resourceDir: string;
+
+  /**
+   * Get the resource directory as used by the API category for GraphQL
+   * @param context the context from the CLI
+   * @param options the options from the CLI
+   */
+  getResourceDir = async (
+    context: $TSContext,
+    options: $TSAny,
```

Comment:
If you only need `resourceDir` from this, can we avoid starting off with $TSAny and accept only what you need for this method?

Reply:
unfortunately no, the underlying types are not exported and are managed by the CLI team. Cleaning up the use of context/options is probably a project in itself, so for the time being I'd prefer the approach of centralizing the usage of context/options like this